File: assignments/a1/q2/Java/Rule.java
//Rule:top//
//Rule:import//
import java.util.*;

// <rule> ::= QUOTE <input> QUOTE
public class Rule /*Rule:class*/ {

    public static final String $className = "Rule";
    public static final String $ruleString =
        "<rule> ::= QUOTE <input> QUOTE";

    public Input input;

    public Rule(Input input) {
//Rule:init//
        this.input = input;
    }

    public static Rule parse(Scan scn$, Trace trace$) {
        if (trace$ != null)
            trace$ = trace$.nonterm("<rule>", scn$.lno);
        scn$.match(Token.Match.QUOTE, trace$);
        Input input = Input.parse(scn$, trace$);
        scn$.match(Token.Match.QUOTE, trace$);
        return new Rule(input);
    }

//Rule//
}

File: assignments/a1/q2/Python/Rule.py
#Rule:top#
#Rule:import#

# <rule> ::= QUOTE <input> QUOTE
class Rule(): #Rule:class#

    className = "Rule"
    ruleString = "<rule> ::= QUOTE <input> QUOTE"
    input = None

    def __init__(input):
        #Rule:init#
        self.input = input

#Rule#
